airbyte-webapp/src/hooks/services/useAdvancedModeSetting.ts

```diff
@@ -0,0 +1,19 @@
+import { useLocalStorage } from "react-use";
+
+import { useCurrentWorkspace } from "hooks/services/useWorkspace";
+
+type SettingsByWorkspace = Record<string, boolean>;
+
+export const useAdvancedModeSetting = (): [boolean, (newSetting: boolean) => void] => {
+  const { workspaceId } = useCurrentWorkspace();
+  const [advancedModeSettingsByWorkspace, setAdvancedModeSettingsByWorkspace] = useLocalStorage<SettingsByWorkspace>(
+    "advancedMode",
+    {}
+  );
+
+  const isAdvancedMode = (advancedModeSettingsByWorkspace || {})[workspaceId] ?? false;
+  const setAdvancedMode = (newSetting: boolean) =>
+    setAdvancedModeSettingsByWorkspace({ ...advancedModeSettingsByWorkspace, ...{ [workspaceId]: newSetting } });
```

Comment:
I think this can be slightly simplified to:
```suggestion
    setAdvancedModeSettingsByWorkspace({ ...advancedModeSettingsByWorkspace, [workspaceId]: newSetting });
```